### PR TITLE
feat(nx-dev): add video course page

### PR DESCRIPTION
**Pull request by juristr**
https://nx-dev-git-video-course-nrwl.vercel.app/courses

### CODE REVIEW

File: nx-dev/ui-courses/src/lib/course-details.tsx [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        <div className="course-title-section">
+          <h1 className="text-2xl font-bold tracking-tight text-slate-950 sm:text-5xl dark:text-white">
+            {course.title}
+          </h1>
+          <div className="mt-4 flex items-center gap-x-2">
+            <BlogAuthors authors={course.authors as BlogAuthor[]} />
+            <span className="text-sm">{course.authors[0].name}</span>
+          </div>
+        </div>
+        {/* <div className="course-icon">
```
Comment (jaysoo) [inline nx-dev/ui-courses/src/lib/course-details.tsx:48]: Remove?

File: nx-dev/ui-courses/src/lib/lesson-player.tsx [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+          </div>
+        </div>
+
+        {/* Center/Right Panel - Video & Content */}
+        <div className="wide:w-[45%] wide:flex-none wide:overflow-y-visible wide:border-r wide:border-slate-200 wide:dark:border-slate-700 flex flex-1 flex-col overflow-y-auto">
+          {/* Video Section */}
+          <div className="flex-none border-b border-slate-200 dark:border-slate-700">
+            {lesson.videoUrl ? (
+              <div className="aspect-video">
+                <YouTube
```
Comment (jaysoo) [inline nx-dev/ui-courses/src/lib/lesson-player.tsx:131]: Is there a way to disable related video at the end of the cource video? Is there a way we can show a CTA to go to the next lesson when the video is stopped? Or we always show a CTA under the video?

File: nx-dev/ui-video-courses/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export * from './lib/course-hero';
+export * from './lib/course-overview';
+// Note: We've removed the export for 'call-to-action' as it's not being used in this context
```
[the remark this answers is not in the document]
Comment (juristr) [reply nx-dev/ui-video-courses/src/index.ts:3]: oops, copy & paste leftover

File: nx-dev/ui-video-courses/src/lib/course-overview.tsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import Link from 'next/link';
+import Image from 'next/image';
```
[the remark this answers is not in the document]
Comment (juristr) [reply nx-dev/ui-video-courses/src/lib/course-overview.tsx:3]: Yeah we might include images on a course page, but then it's a lot of work to create them so I left them out. Agree RE Image component.

File: nx-dev/ui-courses/src/lib/lesson-player.tsx (hunk 2 above)
Comment (juristr) [reply nx-dev/ui-courses/src/lib/lesson-player.tsx:131]: Yeah that's planned for an upcoming PR. I saw it as well and looked into it. YT has an API you can leverage that allows you to get information about the watch progress etc, so you could add an overlay towards the end with a button that leads you to the next lesson. Def want to add this, but wanted to keep this first push minimal and just get it out.

Coded most of it literally on the flight back from MW, so I didn't want to spend too much time on it and then incrementally improve it

File: nx-dev/ui-courses/src/lib/lesson-player.tsx [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+}
+
+export function LessonPlayer({ course, lesson }: LessonPlayerProps) {
+  const [isExpanded, setIsExpanded] = useState(false);
+  const { node: lessonContent } = renderMarkdown(lesson.description, {
+    filePath: lesson.filePath,
+    headingClass: 'scroll-mt-20',
+  });
+
+  const LessonsList = () => (
```
[the remark this answers is not in the document]
Comment (juristr) [reply nx-dev/ui-courses/src/lib/lesson-player.tsx:41]: oh yeah, 100%

File: nx-dev/ui-courses/src/lib/lesson-player.tsx [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+import { Header } from '@nx/nx-dev/ui-common';
+
+interface LessonPlayerProps {
+  course: Course;
+  lesson: Lesson;
+}
+
+// Add this CSS at the top of the file
+const customStyles = `
+  @media (min-width: 1800px) {
```
[the remark this answers is not in the document]
Comment (juristr) [reply nx-dev/ui-courses/src/lib/lesson-player.tsx:18]: seems to work

**Review by mhartington — APPROVED** [review]

**Review by meeroslav — APPROVED** [review]

**Review by jaysoo — APPROVED** [review]

**Review by isaacplmann — APPROVED** [review]

**Pull request merged**